specification/servicefabric/data-plane/Microsoft.ServiceFabric/6.0/servicefabric.json

```diff
@@ -3,7 +3,7 @@
   "info": {
     "title": "Service Fabric Client APIs",
     "description": "Service Fabric REST Client APIs allows management of Service Fabric clusters, applications and services.",
-    "version": "6.0"
+    "version": "6.0.0.1"
```

Comment:
This seems strange. If you are upgrading the version, then you should include the changes in a new folder and create corresponding changes in the readme file. Such changes cannot happen in the same file. Please fix them

Reply:
The minor version of 6.0 is the same. We want to use this to identify what build version of the swagger we have is currently released. Is there an issue with incrementing this version?

Reply:
@amarzavery Please share your thoughts on this

Reply:
This is a bug fix on the same version, we can change the folder but we believe it is not necessary unless we change the major.minor numbers.

Reply:
Until now we have only have YYYY-mm-dd as the format for api-version. This is the first time that we are encountering semantic version in the api-version. I understand that this is data plane and is not governed by ARM. I am ok with following the sematic version rules.
@VipulM-MSFT - Some questions to clear my doubt.
- I saw that the examples still have api-version as 6.0 and not 6.0.0.1. So if the customer sends 6.0 or 6.0.23.0 or 6.0.23.34 how will the service differentiate? Just trying to understand other side of the coin.
- As of today the doc team is showing documentation for the latest version of the swagger spec. However they are working on providing documentation (which includes REST API docs, examples, etc.) for that api-version. So with the semantic versioning model how will they be able to document the delta between different api-versions? Do you see that as a concern for your customers.

We don't want to be a blocker. Just wanted to be sure that we are moving in the right direction.
/cc @johanste

Reply:
@amarzavery How would your questions be different for a team using the date version format?

- Our service looks for a query parameter that indicates the API version to use  
- I'm not sure how this could be a concern. Is the fear that customers may see too many API versions? If people are interested in finding deltas, it seems like a feature the viewing interface should support regardless of version identifier.

Reply:
The specification version and the api-version for the APIs are two different things. Every API have a query parameter that shows which api-version is supported.

Reply:
@samedder, the difference between a date based versioning scheme and semver is that in the prior, there is no ambiguity in the version matching - it is always using an exact match. For semver, you typically provide a range of acceptable versions based on the information carried in the different parts of the version declaration.

In this case, I'm assuming (but I don't know for sure) that if a customer requests version 6.0, the server is allowed to assume that 6.0.23 fulfills that request, but not 6.1.23 or 7.0. Is that a correct assumption?

Reply:
@johanste ah does that mean for other services theres no guarantee of backwards compatibility between date versions?

Your assumption is correct, Vipul mentioned the specific mechanism. We can have say a swagger file version 6.0.23, but the APIs will have a query parameter that indicates to our service that this client is attempting to use our 6.0 API, or say 3.0 API, or any prior version where necessary.

Reply:
@johanste - The api-version parameter for the individual API follows two part semantic versioning scheme and you are right for that it accepts any version that up to the current version of the platform.

Here is the information from our documentation: (https://docs.microsoft.com/en-us/rest/api/servicefabric/sfclient-index) 

"The REST APIs are versioned using the api-version query parameter in the URI. All the REST APIs will always support api-version=MajorVersion.MinorVersion where the Major and Minor version number correspond to the version of the Service Fabric cluster runtime being targeted. For instance, while sending requests to a 6.0 cluster, api-version=6.0 is always guaranteed to work, with the latest contract supported by the REST API. However, requests with api-version=6.0 will fail against a 5.5 cluster. To maintain backward compatibility, the cluster will always support any previously supported api-version. So requests with api-version=1.0 or 2.0 or 3.0 will succeed against a 6.0 cluster. The response output will depend on the contract specified for the specific api-version."

Reply:
@samedder, yes, there is no **gurantee** for backwards compat between the dates. It is in everyone's best interest to keep things as backwards compatible as possible, however. So in practice, if you pick a random Azure service, you'll see that **most** things continue to work as before.

Reply:
@johanste - just to make sure, as I mentioned there is already backward compatibility. The difference is that each individual APIs are versioned and they are backward compatible. 

The reason for changing the last number in the specification is because the specification had bug and not the actual product. Nothing changed in the implementation. We believe that the specification version needs to be different when we fix the bug in the specification, otherwise it is hard to reason about two documents that are versioned with the same value but have different content. I hope we can close this thread with this understanding.